typing_tests/merge-pine-options.ts

```diff
@@ -0,0 +1,502 @@
+
+	const objectExpand = await sdk.models.application.getAll({
+		$expand: { is_for__device_type: {}, owns__device: {} },
+	});
+
+	aNumber = objectExpand[0].is_for__device_type[0].id;
+	aNumberOrUndefined = objectExpand[0].should_be_running__release?.__id;
+	aNumber = objectExpand[0].owns__device[0].id;
+	// TODO: ideally this prop shouldn't be there at all
+	anUndefined = objectExpand[0].owns__release;
+
+	const objectExpandWithSelect = await sdk.models.application.getAll({
+		$expand: {
+			is_for__device_type: {
+				$select: ['slug', 'name'],
+			},
+			owns__device: {},
+		},
+	});
+
+	// @ts-expect-error - test case
+	aNumber = objectExpandWithSelect[0].is_for__device_type[0].id;
+	aString = objectExpandWithSelect[0].is_for__device_type[0].name;
+	aString = objectExpandWithSelect[0].is_for__device_type[0].slug;
+
+	aNumberOrUndefined =
+		objectExpandWithSelect[0].should_be_running__release?.__id;
+	aNumber = objectExpandWithSelect[0].owns__device[0].id;
+	// TODO: ideally this prop shouldn't be there at all
+	anUndefined = objectExpandWithSelect[0].owns__release;
+
+	const objectExpandNestedArray = await sdk.models.application.getAll({
+		$expand: {
+			owns__device: {
+				$expand: ['belongs_to__user'],
+			},
+		},
+	});
+	aNumber = objectExpandNestedArray[0].owns__device[0].belongs_to__user[0].id;
+
+	const objectExpandNested = await sdk.models.application.getAll({
+		$expand: {
+			owns__device: {
+				$expand: {
+					belongs_to__user: {},
+				},
+			},
+		},
+	});
+	aNumber = objectExpandNested[0].owns__device[0].belongs_to__user[0].id;
+
+	const objectExpandNestedWithSelect = await sdk.models.application.getAll({
+		$expand: {
+			owns__device: {
+				$expand: {
+					belongs_to__user: {
+						$select: ['username'],
+					},
+				},
+			},
+		},
+	});
+	aNumber =
+		// @ts-expect-error - test case
+		objectExpandNestedWithSelect[0].owns__device[0].belongs_to__user[0].id;
+
+	const apps = await sdk.models.service.getAllByApplication(3, {
+		$select: 'id',
+	});
+	// @ts-expect-error - test case
+	aString = apps[0].service_name;
+})();
+
+// @ application.getWithDeviceServiceDetails
+await (async () => {
+	const app = await sdk.models.application.getWithDeviceServiceDetails(123);
+	const service = app.owns__device[0].current_services;
+	const test: Equals<
+		typeof service,
+		Record<string, BalenaSdk.CurrentService[]>
+	> = EqualsTrue;
+
+	const appWithExpand =
+		await sdk.models.application.getWithDeviceServiceDetails(123, {
+			$select: 'app_name',
+		});
+	// @ts-expect-error - test case
+	aNumber = appWithExpand.id;
+
+	const appWithNestedExpand =
+		await sdk.models.application.getWithDeviceServiceDetails(123, {
+			$select: 'app_name',
+			$expand: {
+				owns__device: {
+					$expand: {
+						image_install: {
+							$expand: {
+								installs__image: {
+									$select: 'is_stored_at__image_location',
+								},
+							},
+						},
+					},
+				},
+				owns__release: {
+					$select: 'id',
```

Comment:
Let's also add v since it's in the select
```suggestion

aString = appWithNestedExpand.app_name
```